pymc3/distributions/bound.py

```diff
@@ -263,20 +81,39 @@ class Bound:
                     'par2', mu=0.0, sigma=1.0, initval=1.0)
     """
 
-    def __init__(self, distribution, lower=None, upper=None):
+    rv_op = boundrv
+
+    def __new__(cls, *args, **kwargs):
+        transform = kwargs.get("transform", UNSET)
+        base_rv_type = args[1].owner.op.dtype
+        if transform is UNSET and base_rv_type not in ["intX", "int32", "int64"]:
+            kwargs["transform"] = cls.default_transform(kwargs)
+        return super().__new__(cls, *args, **kwargs)
+
+    @classmethod
+    def default_transform(cls, kwargs_dict):
```

Comment:
This seems to duplicate a lot of the logic already implemented in:
https://github.com/pymc-devs/pymc3/blob/9d90c891de5a2f491c51f8311b387f051e27538a/pymc3/distributions/continuous.py#L150

We can try to inherit from it in the current Class or make the `default_transform` in `BoundedContinuous` a `staticmethod` that can be called from other places. 

Also this`Bound` class might still have to be split in `Discrete` and `Continuous` classes, which I think is necessary for some samplers to decide how to deal with the variables. In that case `Bound` can be just a wrapper that returns / instantiates the right type of Variable.